utils/build-script

```diff
@@ -1473,6 +1514,191 @@ details of the setups of other systems or automated environments.""")
+    if args.skip_test_freebsd:
+        platforms_to_skip_test.add(StdlibDeploymentTarget.FreeBSD)
+    if args.skip_test_cygwin:
+        platforms_to_skip_test.add(StdlibDeploymentTarget.Cygwin)
+    if args.skip_test_osx:
+        platforms_to_skip_test.add(StdlibDeploymentTarget.OSX)
+    if args.skip_test_ios_host:
+        platforms_to_skip_test.add(StdlibDeploymentTarget.iOS)
+    if args.skip_test_ios_simulator:
+        platforms_to_skip_test.add(StdlibDeploymentTarget.iOSSimulator)
+    if args.skip_test_tvos_host:
+        platforms_to_skip_test.add(StdlibDeploymentTarget.AppleTV)
+    if args.skip_test_tvos_simulator:
+        platforms_to_skip_test.add(StdlibDeploymentTarget.AppleTVSimulator)
+    if args.skip_test_watchos_host:
+        platforms_to_skip_test.add(StdlibDeploymentTarget.AppleWatch)
+    if args.skip_test_watchos_simulator:
+        platforms_to_skip_test.add(StdlibDeploymentTarget.AppleWatchSimulator)
+
+    platforms_to_skip_test_host = set()
+    if args.skip_test_ios_host:
+        platforms_to_skip_test_host.add(StdlibDeploymentTarget.iOS)
+    if args.skip_test_tvos_host:
+        platforms_to_skip_test_host.add(StdlibDeploymentTarget.AppleTV)
+    if args.skip_test_watchos_host:
+        platforms_to_skip_test_host.add(StdlibDeploymentTarget.AppleWatch)
+
+    # We never allow testing Android, currently.
+    #
+    # FIXME: Allow Android host tests to be enabled/disabled by the build
+    # script.
+    platforms_to_skip_test.add(StdlibDeploymentTarget.Android)
+
+    def get_options_for_target(host):
+        def should_build_stdlib_target(stdlib_target):
+            # If building all targets, this is true.
+            if "all" in args.build_stdlib_deployment_targets:
+                return True
+
+            # Otherwise, only build in the restricted list.
+            if stdlib_target in args.build_stdlib_deployment_targets:
+                return True
+
+            # For cross-tools hosts, always allow their one-and-only
+            # stdlib-target to build.
+            if host != args.host_target and host == stdlib_target:
+                return True
+
+        if host == args.host_target:
+            stdlib_targets = args.stdlib_deployment_targets
+        else:
+            stdlib_targets = [host]
+
+        swift_stdlib_targets = []
+        swift_sdks = set()
+        swift_benchmark_targets = []
+        swift_run_benchmark_targets = []
+        swift_test_targets = []
+        for stdlib_target_name in stdlib_targets:
+            # Get the target object.
+            stdlib_target = StdlibDeploymentTarget.get_target_for_name(
+                stdlib_target_name)
+            if stdlib_target is None:
+                diagnostics.fatal("unknown target: %r" % (stdlib_target_name,))
+
+            # Add the SDK to use.
+            swift_sdks.add(stdlib_target.platform.sdk_name)
+
+            # If we aren't building this target, do nothing else.
```

Comment:
Actually there was support for testing without building.  This would be useful in primarily CI if we start doing distributed testing (build for all targets once, and then copy the artifacts over to multiple machines and parallelize the tests by target).  Since we are not doing it now, and since this use case would be covered by a global flag to the effect of "skip all build steps", I'm fine with dropping this feature for now.